include/pisa/index_types.hpp

```diff
@@ -43,12 +46,220 @@ using block_simple8b_index = block_freq_index<pisa::simple8b_block>;
+            return "single";
+        } else if constexpr (std::is_same_v<I, pefuniform_index>) {
+            return "pefuniform";
+        } else if constexpr (std::is_same_v<I, pefopt_index>) {
+            return "pefopt";
+        } else if constexpr (std::is_same_v<I, block_optpfor_index>) {
+            return "block_optpfor";
+        } else if constexpr (std::is_same_v<I, block_varintg8iu_index>) {
+            return "block_varintg8iu";
+        } else if constexpr (std::is_same_v<I, block_streamvbyte_index>) {
+            return "block_streamvbyte";
+        } else if constexpr (std::is_same_v<I, block_maskedvbyte_index>) {
+            return "block_maskedvbyte";
+        } else if constexpr (std::is_same_v<I, block_interpolative_index>) {
+            return "block_interpolative";
+        } else if constexpr (std::is_same_v<I, block_qmx_index>) {
+            return "block_qmx";
+        } else if constexpr (std::is_same_v<I, block_varintgb_index>) {
+            return "block_varintgb";
+        } else if constexpr (std::is_same_v<I, block_simple8b_index>) {
+            return "block_simple8b";
+        } else if constexpr (std::is_same_v<I, block_simple16_index>) {
+            return "block_simple16";
+        } else if constexpr (std::is_same_v<I, block_simdbp_index>) {
+            return "block_simdbp";
+        } else if constexpr (std::is_same_v<I, profiling_block_optpfor_index>) {
+            return "block_optpfor";
+        } else if constexpr (std::is_same_v<I, profiling_block_varintg8iu_index>) {
+            return "block_varintg8iu";
+        } else if constexpr (std::is_same_v<I, profiling_block_streamvbyte_index>) {
+            return "block_streamvbyte";
+        } else if constexpr (std::is_same_v<I, profiling_block_maskedvbyte_index>) {
+            return "block_maskedvbyte";
+        } else if constexpr (std::is_same_v<I, profiling_block_interpolative_index>) {
+            return "block_interpolative";
+        } else if constexpr (std::is_same_v<I, profiling_block_qmx_index>) {
+            return "block_qmx";
+        } else if constexpr (std::is_same_v<I, profiling_block_varintgb_index>) {
+            return "block_varintgb";
+        } else if constexpr (std::is_same_v<I, profiling_block_simple8b_index>) {
+            return "block_simple8b";
+        } else if constexpr (std::is_same_v<I, profiling_block_simple16_index>) {
+            return "block_simple16";
+        } else if constexpr (std::is_same_v<I, profiling_block_simdbp_index>) {
+            return "block_simdbp";
+        }
+    }
+
+    template <typename Index>
+    struct add_profiling {
+        using type = Index;
+    };
+
+    template <typename Block>
+    struct add_profiling<block_freq_index<Block, false>> {
+        using type = block_freq_index<Block, true>;
+    };
+
+    template <bool Profile, std::size_t I, typename Fn>
+    auto resolve_index_n(std::string_view encoding, Fn fn) -> bool
```

Comment:
Why is this called `resolve_index_n`, am I missing something?

Reply:
Well, I suppose it's a big inconsistent. What this one does is it attempts to resolve the encoding at position `I` (template parameter), and returns `true` if it succeeds. I guess it would be clearer if I used `N` instead. I'll also add some docs to all of these today.